### PR TITLE
Update CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by adizere**
Updated the contact email to an `informal.systems` one, and some nits.

@greg-szabo can we allocate a new email address `conduct@informal.systems`, with copied messages to HR, COO, and CometBFT Project Lead?

@thanethomson not sure I'm overdoing it with backport, please correct if you see something off.

---

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [ ] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ hence to CometBFT.
 
 * Private harassment is also unacceptable. No matter who you are, if you feel
   you have been or are being harassed or made uncomfortable by a community
-  member, please contact one of the channel admins or the person mentioned above
+  member, please contact one of the channel admins or the contat address above
```
Comment (melekes) [inline CODE_OF_CONDUCT.md:38]:
```suggestion
  member, please get in touch with one of the channel admins or the contact address above
```

Comment (adizere) [reply CODE_OF_CONDUCT.md]: Nice catch Anton!

**Review by thanethomson — APPROVED** [review]
LGTM, and it's a good idea to backport since the README is rendered to `pkg.go.dev` such that it links to the code of conduct at the relevant tagged release: https://pkg.go.dev/github.com/cometbft/cometbft#section-readme

Speaking of which, we should try to keep the READMEs and other root docs in sync across all backport branches due to the above.

**Review by melekes — APPROVED** [review]

**Review by lasarojc — APPROVED** [review]

**Comment by adizere** [comment]
Blocked on @greg-szabo confirming that `conduct@informal.systems` is good to go.

**Comment by greg-szabo** [comment]
Confirmed.

**Comment by adizere** [comment]
Thank you Greg! Waiting for various members CometBFT community to also have a quick look, offer potential suggestions for improvements, then we'll merge.

**Pull request merged**